### DISCUSSION TITLE
Having "Prompts" and "Documents collection" available for user (not only admin)

### DISCUSSION THREAD

**Discussion by rsochard** [open-webui/open-webui#2360] [General]
**Is your feature request related to a problem? Please describe.**
Actually "Prompts" and "Collection of Documents" are only managed by the administrator, which is (I think) too much restrectied.

**Describe the solution you'd like**
It should be very interested to have those features also availble for the users, so they can manage their own "Prompts" and "Documents" in a private way, without any global sharing between the users.

In a second time, it should be interested to add a feature of "sharing" Prompts & Documents to other dedicated/choosen users.

**Comment by kha84** [comment]
Don't know about prompts - to me it's more like a cherry on a pie. But RAG is a killer feature and it's sad it's not available / not configurable to make it available to just "users", not only "admins"

**Comment by rsochard** [comment]
If the feature is accepted, I already have the code modifications on my side to support it, so I can easily propose something.

With my propositions, the 'Modelfiles', 'Documents' and 'Prompts' can be managed by the "users" and are visible only for them (according to their userid) : i.e. no sharing ressources like what is done for the "admin"

**Reply by H9990HH969** [reply]
Hi rsochard, may I ask if the modified code can be shared? I am also looking for a chat system that supports users managing their own knowledge base.

**Reply by tekos-flow** [reply]
@rsochard Hi, yes, that would be great if you can share this code with the community 🙏

**Reply by dannykorpan** [reply]
Share please :-D Thank you!

**Comment by rsochard** [comment]
Documents are actually stored in a docker volume.
It should be better to have them stored directly in the database (as a blob in Postgres).
This will help to faciliate the scaling of open-web-ui.

**Reply by OM-EL** [reply]
I agree with this , making a POC inside my company right now , this is one of the issues that might stop us from scaling using the same project.

**Reply by OM-EL** [reply]
@rsochard  do you have any idea is this changed ?

**Comment by flefevre** [comment]
Is it preferable to have an additional role such as 'manager'
We have admin, user and manager that are to create workspace?

I do agree it is preferable to store in a separate database rather than in a docker volume.
Each workspace should have its own database configuration for security purposes

**Reply by justinh-rahb** [reply]
It's on the ideas list for the eventual **Teams** feature.

**Comment by drmbt** [comment]
did this discussion end in any conclusions or feature integrations? I want to be able to share prompts, knowledge and workspaces between user groups
